lib/galaxy/tool_util/toolbox/base.py

```diff
@@ -661,6 +661,7 @@ def _load_integrated_tool_panel_keys(self):
                 section = ToolSection(elem)
                 for section_elem in elem:
                     section_id = section_elem.get("id")
+                    assert section_id
```

Comment:
```suggestion
                    assert section_id, f"Element '{etree.tostring(section_element)}' did not specify 'id' attribute"
```
this seems like it could really happen